chainerrl/replay_buffer.py

```diff
@@ -135,35 +143,44 @@ def __init__(self, capacity=None, num_steps=1):
         assert num_steps > 0
         self.num_steps = num_steps
         self.memory = RandomAccessQueue(maxlen=capacity)
-        self.last_n_transitions = collections.deque([], maxlen=num_steps)
+        self.last_n_transitions = collections.defaultdict(
+            lambda: collections.deque([], maxlen=num_steps))
 
     def append(self, state, action, reward, next_state=None, next_action=None,
-               is_state_terminal=False):
-        experience = dict(state=state, action=action, reward=reward,
-                          next_state=next_state, next_action=next_action,
-                          is_state_terminal=is_state_terminal)
-        self.last_n_transitions.append(experience)
+               is_state_terminal=False, env_id=0, **kwargs):
+        last_n_transitions = self.last_n_transitions[env_id]
+        experience = dict(
+            state=state,
+            action=action,
+            reward=reward,
+            next_state=next_state,
+            next_action=next_action,
+            is_state_terminal=is_state_terminal,
+            **kwargs
+        )
+        last_n_transitions.append(experience)
         if is_state_terminal:
```

Comment:
Why do we remove the if statement that checks if the transition is terminal?

Reply:
i think it is not removed